cli/cmd/root.go

```diff
@@ -100,6 +108,7 @@ func init() {
 	RootCmd.PersistentFlags().StringArrayVar(&impersonateGroup, "as-group", []string{}, "Group to impersonate for Kubernetes operations")
 	RootCmd.PersistentFlags().StringVar(&apiAddr, "api-addr", "", "Override kubeconfig and communicate directly with the control plane at host:port (mostly for testing)")
 	RootCmd.PersistentFlags().BoolVar(&verbose, "verbose", false, "Turn on debug logging")
+	RootCmd.PersistentFlags().StringVar(&cliVersionOverride, "cli-version-override", "", "Used to override the version of the cli (mostly for testing)")
```

Comment:
WDYT about exposing this option only to `linkerd check`, alongside `--expected-version` instead of making it a global flag?